File: LeetCode/1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.length(), m = word2.length();
        int i = 0, j = 0;
        string ans = "";
        while (i < n && j < m) {
            ans += word1[i];
            i++;
            ans += word2[j];
            j++;
        }
        while (i < n) {
            ans += word1[i];
            i++;
        }
        while (j < m) {
            ans += word2[j];
            j++;
        }
        return ans;
    }
};

File: LeetCode/1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder ans = new StringBuilder();
        int n = word1.length(), m = word2.length();
        int i = 0, j = 0;
        while (i < n || j < m) {
            if (i < n) {
                ans.append(word1.charAt(i));
                i++;
            }
            if (j < m) {
                ans.append(word2.charAt(j));
                j++;
            }
        }
        return ans.toString();
    }
}